src/base/Algorithms.hh

```diff
@@ -48,6 +48,18 @@ CELER_CONSTEXPR_FUNCTION T ipow(T v)
                           : v * ipow<(N - 1) / 2>(v) * ipow<(N - 1) / 2>(v);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Interchange (swap) the values of two inputs.
+ */
+template<typename T>
+CELER_CONSTEXPR_FUNCTION void swap2(T& v1, T& v2)
+{
+    T tmp = v1;
+    v1    = v2;
+    v2    = tmp;
+}
+
```

Comment:
A new specialization of swap on `Quantity` means we can delete this, see a comment below where this is used.